cyral/internal/policyv2/datasource.go

```diff
@@ -0,0 +1,112 @@
+package policyv2
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/cyralinc/terraform-provider-cyral/cyral/client"
+	"github.com/cyralinc/terraform-provider-cyral/cyral/core"
+	"github.com/cyralinc/terraform-provider-cyral/cyral/core/types/resourcetype"
+)
+
+var dsContextHandler = core.DefaultContextHandler{
+	ResourceName:                 dataSourceName,
+	ResourceType:                 resourcetype.DataSource,
+	SchemaWriterFactoryGetMethod: func(_ *schema.ResourceData) core.SchemaWriter { return &PolicyV2{} },
+	ReadUpdateDeleteURLFactory: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf("https://%s/%s/%s", c.ControlPlane, getAPIPath(d.Get("type").(string)), d.Get("id").(string))
+	},
+}
+
+func dataSourceSchema() *schema.Resource {
+	return &schema.Resource{
+		Description: "Some description.",
+		ReadContext: dsContextHandler.ReadContext(),
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Description: "Retrieve the unique label with this name, if it exists.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"description": {
+				Description: "Description of the data source.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"enabled": {
+				Description: "Indicates if the policy is enabled.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"scope": {
+				Description: "Repos that the policy applies to, empty value implies applicability to all repos.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"tags": {
+				Description: "Tags associated with the policy to categorize it.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"valid_from": {
+				Description: "Time when the policy comes into effect.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"valid_until": {
+				Description: "Time after which the policy is no longer in effect.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"document": {
+				Description: "The actual policy document in JSON format.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"last_updated": {
+				Description: "Information about when and by whom the policy was last updated.",
+				Type:        schema.TypeMap,
+				Optional:    true,
```

Comment:
This should be:
```suggestion
				Computed:    true,
```

Reply:
thanks; 4526ad68ec5dd5249aedaa4fcfc0a8f9c53db044